compiler/rustc_lint/src/context.rs

```diff
@@ -220,17 +220,17 @@ impl LintStore {
                         })
                         .lint_ids
                         .push(id);
+                } else {
+                    self.lint_groups
```

Comment:
It'd be nice to have a comment explaining the logic here. Something like:

"Future incompatible" lints are lints where a future version of rustc will cause the *existing code to stop compiling*. Edition lints don't count because they require "opt-in".